pkg/operatorsource/configuring.go

```diff
@@ -95,20 +131,86 @@ func (r *configuringReconciler) Reconcile(ctx context.Context, in *marketplace.O
 	}
 
 	// If we are here, the given CatalogSourceConfig object already exists.
-	cscNamespacedName := types.NamespacedName{Name: in.Name, Namespace: in.Namespace}
-	cscExisting := marketplace.CatalogSourceConfig{}
-	err = r.client.Get(ctx, cscNamespacedName, &cscExisting)
+	err = r.updateExistingCatalogSourceConfig(ctx, in, packages)
 	if err != nil {
-		r.logger.Errorf("Unexpected error while getting CatalogSourceConfig: %s", err.Error())
+		r.logger.Errorf("Unexpected error while updating CatalogSourceConfig: %s", err.Error())
 		nextPhase = phase.GetNextWithMessage(phase.Configuring, err.Error())
-
 		return
 	}
 
+	r.logger.Info("CatalogSourceConfig object has been updated successfully")
+
+	nextPhase = phase.GetNext(phase.Succeeded)
+	return
+}
+
+// getManifestMetadata gets the package metadata from the OperatorSource endpoint
+// it returns the list of packages to be written to the OperatorSource status and an error
+func (r *configuringReconciler) getManifestMetadata(spec *marketplace.OperatorSourceSpec, namespace string) ([]*datastore.RegistryMetadata, error) {
+
+	manifests := make([]*datastore.RegistryMetadata, 0)
+
+	options, err := SetupAppRegistryOptions(r.client, spec, namespace)
+	if err != nil {
+		return manifests, err
+	}
+
+	registry, err := r.factory.New(options)
+	if err != nil {
+		return manifests, err
+	}
+
+	manifests, err = registry.ListPackages(spec.RegistryNamespace)
+	if err != nil {
+		return manifests, err
+	}
+
+	return manifests, nil
+}
+
+// checkDatastoreForExistingPackages checks to see if there are any packages before we write to the datastore
+// If there are not, we are assuming this is a new operator source. We should force all CatalogSourceConfig
+// to compare their versions to what's in the datastore after we update it.
+// the function returns the count of the packages in the datastore and an error
+func (r *configuringReconciler) checkDatastoreForExistingPackages(in *marketplace.OperatorSource, out *marketplace.OperatorSource, manifests []*datastore.RegistryMetadata) (int, error) {
+
+	preUpdateDatastorePackageList := r.datastore.GetPackageIDsByOperatorSource(out.GetUID())
+
+	count, err := r.datastore.Write(in, manifests)
+	if err != nil {
+		if count == 0 {
+			return count, err
+		}
+		// Ignore faulty operator manifests
+		r.logger.Infof("There were some faulty operator manifest(s), errors - %v", err)
+		err = nil
+	}
+
+	// Now that we have updated the datastore, let's check if the opsrc is new.
```

Comment:
The function is called `checkDataStoreForExistingPackages`, which it does, but also does the job of forcing a resync of the `CatalogSourceConfig`. Maybe we can separate the two jobs or name the function something else that capture the purpose of the function in it's entirety?

Reply:
I see your point but I think it makes sense to have all the syncing done in this function. Maybe @aravindhp or @kevinrizza have an opinion? Or we could change the name to show this is happening here?

Reply:
I see @anik120's point. This function is actually writing to the datastore and deciding if a sync is needed. We could separate those concerns.

Reply:
That makes sense - thanks for noticing this @anik120 I'll push a change